src/Lykke.Job.TransactionHandler/Handlers/EthereumCoreCommandHandler.cs

```diff
@@ -0,0 +1,468 @@
+            var asset = await _assetsServiceWithCache.TryGetAssetAsync(token.AssetId);
+            var amount = EthServiceHelpers.ConvertFromContract(queueMessage.Amount, asset.MultiplierPower, asset.Accuracy);
+
+            await HandleCashInOperation(asset, amount, bcnCreds.ClientId,
+                bcnCreds.Address, queueMessage.TransactionHash, eventPublisher);
+
+            return true;
+        }
+
+        private async Task<bool> ProcessHotWalletCashout(ProcessHotWalletEventCommand queueMessage)
+        {
+            string transactionId = queueMessage.OperationId;
+            CashOutContextData context = await _transactionService.GetTransactionContext<CashOutContextData>(transactionId);
+            string clientId = context.ClientId;
+            string hash = queueMessage.TransactionHash;
+            string cashOperationId = context.CashOperationId;
+
+            await _cashOperationsRepositoryClient.UpdateBlockchainHashAsync(clientId, cashOperationId, hash);
+
+            return true;
+        }
+
+        private async Task<bool> ProcessOutcomeOperation(ProcessCoinEventCommand queueMessage)
+        {
+            var transferTx = await _ethereumTransactionRequestRepository.GetAsync(Guid.Parse(queueMessage.OperationId));
+            CashOutContextData context = await _transactionService.GetTransactionContext<CashOutContextData>(queueMessage.OperationId);
+
+
+            if (transferTx != null)
+            {
+                switch (transferTx.OperationType)
+                {
+                    case OperationType.CashOut:
+                        await SetCashoutHashes(transferTx, queueMessage.TransactionHash);
+                        break;
+                    case OperationType.Trade:
+                        await SetTradeHashes(transferTx, queueMessage.TransactionHash);
+                        break;
+                    case OperationType.TransferToTrusted:
+                    case OperationType.TransferFromTrusted:
+                        await SetTransferHashes(transferTx, queueMessage.TransactionHash);
+                        break;
+                }
+
+                return true;
+            }
+
+            if (context != null)
+            {
+                string clientId = context.ClientId;
+                string hash = queueMessage.TransactionHash;
+                string cashOperationId = context.CashOperationId;
+
+                await _cashOperationsRepositoryClient.UpdateBlockchainHashAsync(clientId, cashOperationId, hash);
+            }
+
+
+            return true;
+        }
+
+        //TODO: Split wth the help of the process management
+        private async Task<bool> ProcessFailedCashout(ProcessCoinEventCommand queueMessage, IEventPublisher eventPublisher)
+        {
+            CashOutContextData context = await _transactionService.GetTransactionContext<CashOutContextData>(queueMessage.OperationId);
+
+            if (context != null)
```

Comment:
Inverting of this condition will reducre method nesting

Reply:
fixed